File: lib/src/data/telegram_repository.dart
import 'package:flutter/foundation.dart';

import '../models/telegram_models.dart';

abstract class TelegramRepository extends ChangeNotifier {
  AuthState get authState;
  List<ChatDialog> get dialogs;

  List<ChatMessage> messagesFor(String chatId);

  Future<void> startLogin(LoginCredentials credentials);
  Future<void> submitCode(String code);
  Future<void> submitPassword(String password);
  Future<void> loadMessages(String chatId);
  Future<void> sendText({required String chatId, required String text});
  Future<void> logout();
}

File: core/src/main/kotlin/com/sing/tgthird/core/repository/TelegramRepository.kt
package com.sing.tgthird.core.repository

import com.sing.tgthird.core.model.AuthState
import com.sing.tgthird.core.model.ChatDialog
import com.sing.tgthird.core.model.LoginCredentials
import com.sing.tgthird.core.model.Message
import kotlinx.coroutines.flow.Flow
import kotlinx.coroutines.flow.StateFlow

interface TelegramRepository {
    val authState: StateFlow<AuthState>
    val dialogs: StateFlow<List<ChatDialog>>

    fun messagesFor(chatId: String?): Flow<List<Message>>
    suspend fun startLogin(credentials: LoginCredentials)
    suspend fun submitCode(code: String)
    suspend fun submitPassword(password: String)
    suspend fun sendText(chatId: String, text: String)
    suspend fun logout()
}
